### PR TITLE
Fix parameter access from solution

### PR DESCRIPTION
**Pull request by ValentinKaisermayer**

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Needs tests

**Comment by ValentinKaisermayer** [comment]
I think all of the solution interfaces need a refactor there is so much code duplication, that it makes it really hard to maintain.

**Comment by ChrisRackauckas** [comment]
It is undergoing a heavy refactor right now in the generalization via SymbolicIndexingInterface.jl (https://github.com/SciML/SciMLBase.jl/pull/342). We should finish the generalization and test it all downstream well before starting the refactor. One of the things that will need a heavy refactor is the plot recipes. But yes, now that things can hold and query systems, all of that should only use the SymbolicIndexingInterface.jl functions and the fallbacks should be deprecated.

File: test/downstream/Manifest.toml [hunk 1]
```diff
@@ -0,0 +1,1434 @@
+# This file is machine-generated - editing it directly is not advised
```
Comment (ChrisRackauckas) [inline test/downstream/Manifest.toml:1]: no manifest

Comment (ValentinKaisermayer) [reply test/downstream/Manifest.toml]: ups, sry

**Pull request merged**